lisp/ess-r-package.el

```diff
@@ -268,6 +273,24 @@ arguments, or expressions which return R arguments."
         args
       (concat ", " args))))
 
+(defun ess-r-package-compilation-start (command)
+  "Run `compilation-start' in the package's root directory."
+  (let* ((pkg-info (or (ess-r-package-project)
+                       (ess-r-package-set-package)))
+         (pkg-name (ess-r-package-name))
+         (default-directory (cdr pkg-info)))
+    (compilation-start command nil
+                       (lambda (name-of-mode)
+                         (concat "*" (downcase name-of-mode) ":" pkg-name "*"))
+                       ess-r-error-regexp-alist)))
+
+(defun ess-r-package-eval-in-compile-buffer (expr)
+  "Evalate R expression EXPR in a standalone `compilation-mode'
+buffer."
+  (let* ((procname inferior-ess-r-program-name))
+    (ess-r-package-compilation-start
```

Comment:
I think this extra helper is not warranted. Could be inlined here.

Reply:
Indeed. Let's strive for minimal code for now.

Reply:
Fixed.